connector-j-8/src/main/java/com/google/cloud/sql/mysql/SocketFactory.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.sql.mysql;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.cloud.sql.core.SslSocketFactory;
+import com.mysql.cj.protocol.ServerSession;
+import com.mysql.cj.protocol.SocketConnection;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.net.Socket;
+import java.util.Properties;
+import java.util.logging.Logger;
+import jnr.unixsocket.UnixSocketAddress;
+import jnr.unixsocket.UnixSocketChannel;
+
+/**
+ * A MySQL {@link SocketFactory} that establishes a secure connection to a Cloud SQL instance using
+ * ephemeral certificates.
+ *
+ * <p>The heavy lifting is done by the singleton {@link SslSocketFactory} class.
+ */
+public class SocketFactory implements com.mysql.cj.protocol.SocketFactory {
```

Comment:
Do you have a documentation link to the `com.mysql.cj.protocol.SocketFactory`?

Reply:
Unfortunately I've found no documentation of the connector's overriding classes like this one or `QueryInterceptor` :(